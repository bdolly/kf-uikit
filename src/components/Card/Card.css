.Card {
    @apply 
        inline-block
        overflow-hidden 
        max-w-sm
        w-full
        m-2 
        p-4
        bg-white
        rounded-sm
        border
        border-grey-lighter
        text-base
        font-normal;

    hr {
        border-style: solid;
        @apply
            border
            border-grey-lighter;
    }
}

.Card--title {
    @apply
        m-0
        text-secondary;
}
