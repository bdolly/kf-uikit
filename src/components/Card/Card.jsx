import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';

/**
 * A Simple card
 */
const Card = ({ className, title, children }) => {

  const cardClass = classes(
    'Card',
    className,
  );

  return (
    <div className={cardClass}>
      {React.isValidElement(title) ? (
        <span className='Card--title'>{title}</span>
      ) : (
        <h2 className='Card--title text-truncate' title={title}>{title}</h2>
      )}
      <hr />
      {children}
    </div>
  );
}

Card.propTypes = {
  /** Additonal css classes to be applied to the card */
  className: propTypes.string,
  /** Title of the card */
  title: propTypes.oneOfType([propTypes.element, propTypes.string]),
  /** Child elements */
  children: propTypes.node,
};

Card.defaultProps = {
  className: null,
  title: null,
  children: null,
};

export default Card;
