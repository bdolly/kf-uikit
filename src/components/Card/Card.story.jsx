/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Card from './Card';

const stories = storiesOf('Cards', module);

stories.add('Simple Card', () => (
  <Card title={text('title', 'A very very long card title lorem ipsum dolor')}>
    <p> Lorem ipsum </p>
  </Card>
));

stories.add('Many Cards', () => (
  <div className="flex">
    <Card className="flex-1" title="Card 1">
      <p> Lorem ipsum </p>
    </Card>
    <Card className="flex-1" title="Card 2">
      <p> Lorem ipsum </p>
    </Card>
  </div>
));
