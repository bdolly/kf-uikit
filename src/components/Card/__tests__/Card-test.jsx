import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';

['Testing', <em>Test</em>].map((title) => {
  it(`${title} Card renders correctly`, () => {
      const tree = renderer.create(
              <Card title={title}>
                Hello
              </Card>
              ).toJSON();
        expect(tree).toMatchSnapshot();
  });
});
