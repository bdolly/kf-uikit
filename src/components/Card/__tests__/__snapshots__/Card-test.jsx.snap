// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] Card renders correctly 1`] = `
<div
  className="Card"
>
  <span
    className="Card--title"
  >
    <em>
      Test
    </em>
  </span>
  <hr />
  Hello
</div>
`;

exports[`Testing Card renders correctly 1`] = `
<div
  className="Card"
>
  <h2
    className="Card--title text-truncate"
    title="Testing"
  >
    Testing
  </h2>
  <hr />
  Hello
</div>
`;
